### ISSUE TITLE
ngx-datatable-row-detail collapsing by itself

### ISSUE THREAD

**Issue by lucasctd** [swimlane/ngx-datatable#1640]
**I'm submitting a ...**
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, post on Stackoverflow or Gitter
```

**Current behavior**

When I select an input inside `<ngx-datatable-row-detail>` the detail will collapse. If I expand it again and select a value from a `<select>` or edit an `<input>` it will collapse again.

**Expected behavior**
The detail should stay expanded.

**Reproduction of the problem**
Add `<input>` or `<select>` elements inside `<ngx-datatable-row-detail>` and try to interact with them.
You can [watch](https://youtu.be/IOOpCV-ElJc) the unexpected behaviour [here](https://youtu.be/IOOpCV-ElJc) too.

**What is the motivation / use case for changing the behavior?**

**Please tell us about your environment:**

* **OS:** 
MS Windows 10 Pro

* **IDE:** 
VS Code/WebStorm

* **Package Manager:** 
NPM

* **Table version:** 
14.0.0

* **Angular version:** 
6.1.10

* **Browser:**
Chrome, Opera, Chrome Mobile

* **Language:** [TypeScript 2.9.2 | ES6/7]

**Comment by lucasctd** [comment]
Btw, I've tried
```
(focusout)="$event.stopImmediatePropagation()"
(blur)="$event.stopImmediatePropagation()"
(mouseup)="$event.stopImmediatePropagation()"
(click)="$event.stopImmediatePropagation()"
(change)="$event.stopImmediatePropagation()"
```
with no success.

**Comment by rubenfig** [comment]
Try with                 
(select)="$event.stopPropagation()"

It worked for me.

**Comment by lucasctd** [comment]
It didn't work :v I really don't know how the table knows I am interacting with the inputs since I have stopped the propagation of most known events.

**Comment by Alvol** [comment]
@lucasctd can also try both
 ` event.stopPropagation();
   event.preventDefault();`
if it will not help try to check `(activate)` output of ngx-datatable itself, maybe you will find what trigger your toggle event.

**Comment by SeniorCSSdeveloper** [comment]
@lucasctd Have you figured this issue out yet? I'm facing the same problem. My row-detail collapses on server-side request.

**Comment by lucasctd** [comment]
@SeniorCSSdeveloper Nope, I am working on the backend of the application now. I still have to fix this but I have other priorities, when I got back to work on it, I will post the solution here if I found.

**Comment by ermarkar** [comment]
@lucasctd you got any solution for this?

**Comment by ali9itani** [comment]
I am facing a similar issue.
 When I use mat-select or ng-select, the 'row details' is not opening.
The solution to this was to use the native select as follows:

`<select (change)="onSelect($event)">` 
`  <option>1</option>` 
`  <option>2</option>` 
`</select>` 

and then I can bind to the value or get it on select

`
onSelect($event) {
  console.log($event.target.value);
}
`

**Comment by KokalaBhanusri** [comment]
did any one get any solution for this?

**Comment by lucasctd** [comment]
@ermarkar @KokalaBhanusri Hi, I am sorry for taking too long to answer. I actually haven't figured out how to fix this.
